scheduler/scheduler.go

```diff
@@ -11,16 +12,27 @@ import (
 // be handled via a config file in a prod system.
 const (
 	grpcServerAddr    = "127.0.0.1:50000"
-	grpcServerUseTLS  = false
-	grpcServerCrtFile = "server.pem"
-	grpcServerKeyFile = "server.key"
+	grpcServerUseTLS  = true
+	grpcServerCrtFile = "scheduler.crt"
+	grpcServerKeyFile = "scheduler.key"
+
+	defAPIKey = "default-api-key"
 )
 
 var (
 	errs = make(chan error)
 	sig  = make(chan os.Signal)
+
+	apiKey string
 )
 
+func init() {
+	flag.StringVar(&apiKey, "api_key", defAPIKey, "API key for the worker and scheduler api communication.")
```

Comment:
Looks like you have a single global API key? The plan was to have the ability to register workers right? That way a single compromised key does not compromise all?